src/components/common/table/index.tsx

```diff
@@ -1,53 +1,87 @@
 import ActionCell from './cells/actionCell';
+import CustomModal from '../../modals/customModal';
 import CustomToolbar from './customToolbar';
+import { GridColDef } from '@mui/x-data-grid';
+import { IContactState } from '../../../types/contacts';
 import { ITicketState } from '../../../types/tickets';
-import TicketsModal from '../../modals/tickets';
-import { deleteTicketAsync } from '../../../store/tickets/thunk';
+import { RootState } from '../../../store';
 import { useAppDispatch } from '../../../store/hooks';
 import { useSelector } from 'react-redux';
-import { useTableSortAndFilter } from './useTableSort';
+import { DeleteAsyncThunk, FetchAthynkThunk, useTableSortAndFilter } from './useTableSortAndFilter';
 import { StyledBox, StyledDataGrid } from './styled';
-import { baseMenuCellConfig, columns, pageSizeOptions } from './helper';
-import { tickets, total } from '../../../store/tickets/selector';
+import { baseMenuCellConfig, pageSizeOptions } from './helper';
 import { useCallback, useState } from 'react';
 
-const TicketsTable = () => {
+export interface ISortingOptions {
+  label: string;
+  value: string;
+}
+
+export enum EFormType {
+  Contacts = 'Contacts',
+  Tickets = 'Tickets',
+}
+type DataSelector = (state: RootState) => {
+  storeItems: ITicketState[] | IContactState[];
+  storeTotal: number;
+};
+
+interface IProps {
+  storeData: DataSelector;
+  fetchAction: FetchAthynkThunk;
+  deleteAction: DeleteAsyncThunk;
+  columns: GridColDef[];
+  sortingOptions: ISortingOptions[];
+  priorityOptions?: string[];
+  disabledFilter?: boolean;
+  formType: EFormType;
+}
+
+const Table: React.FC<IProps> = ({
+  storeData,
+  fetchAction,
+  deleteAction,
+  columns,
+  sortingOptions,
+  priorityOptions,
+  disabledFilter,
+  formType,
+}) => {
   const dispatch = useAppDispatch();
 
+  const { storeItems, storeTotal } = useSelector(storeData);
+
   const {
     searchParams,
     setSelectedPriorities,
     selectedPriorities,
     paginationModel,
     setPaginationModel,
-  } = useTableSortAndFilter();
+  } = useTableSortAndFilter(fetchAction);
 
   const [isModalOpen, setIsModalOpen] = useState(false);
-  const [selectedTicket, setSelectedTicket] = useState<ITicketState | null>(null);
-
-  const storeTickets = useSelector(tickets);
-  const storeTotal = useSelector(total);
+  const [selectedItem, setSelectedItem] = useState<ITicketState | IContactState | null>(null);
 
   const handleUpdateItem = useCallback(
     (id?: string) => {
-      const ticketToEdit = storeTickets?.find((ticket) => ticket.id === id) || null;
-      setSelectedTicket(ticketToEdit);
+      const itemToEdit =
+        (storeItems as (ITicketState | IContactState)[])?.find((item) => item.id === id) || null;
```

Comment:
I think that you can use if(storeItems && storeItems.length)

Reply:
No, it doesnt work.